homework/index.js

```diff
@@ -29,19 +29,124 @@
     });
     return elem;
   }
+  // table of repository's basic information
+  function createDetailsTable(index, data, container) {
+    const repositoryData = createAndAppend('div', container, { class: 'left-div white-frame' });
+    const table = createAndAppend('table', repositoryData);
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    // manually create td with a link
+    const td = createAndAppend('td', tr);
+    createAndAppend('a', td, {
+      href: data[index].html_url,
+      text: data[index].name,
+      target: '_blank',
+    });
+    // a function to create td without link
+    function createTd(textKey, textContent) {
+      const parent = createAndAppend('tr', tbody);
+      createAndAppend('td', parent, {
+        text: `${textKey}: `,
+        class: 'label',
+      });
+      createAndAppend('td', parent, {
+        text: textContent,
+      });
+    }
+    createTd('Description', data[index].description);
+    createTd('Forks', data[index].forks);
+    createTd('Updated', data[index].updated_at);
+  }
+
+  // contributors list
+  function listContributors(index, data, container) {
+    const contributorUrl = data[index].contributors_url;
+    fetchJSON(contributorUrl, (err, contributors) => {
+      const root = document.getElementById('root');
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
```

Comment:
When dealing with errors like this at the start of the function body, it is best to 'exit early': there is no more useful work to do in the function:

```js
if (err) {
  createAndAppend('div', root, { text: err.message, class: 'alert-error' });
  return;
}

const contributorsData = createAndAppend('div', container, {
...
```

You can now also get rid of the `else` and the corresponding indentation for the rest of the function body.

You can do the same in lines 136-137.

Reply:
It works the same way, no need for the `else`.
Thanks for the tip.